packages/sensors_plus/tizen/src/sensors_plus_plugin.cc

```diff
@@ -67,40 +73,129 @@ class SensorsPlusPlugin : public flutter::Plugin {
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/user_accel",
             &flutter::StandardMethodCodec::GetInstance());
-    user_accel_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(SensorType::kUserAccel));
+    std::unique_ptr<DeviceSensor> user_accelerometer_sensor =
+        std::make_unique<DeviceSensor>(SensorType::kUserAccel, interval_ms_);
+    user_accelerometer_sensor_ = user_accelerometer_sensor.get();
+    user_accelerometer_channel->SetStreamHandler(
+        std::make_unique<DeviceSensorStreamHandler>(
+            std::move(user_accelerometer_sensor)));
 
     std::unique_ptr<FlEventChannel> magnetometer_channel =
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/magnetometer",
             &flutter::StandardMethodCodec::GetInstance());
+    std::unique_ptr<DeviceSensor> magnetometer_sensor =
+        std::make_unique<DeviceSensor>(SensorType::kMagnetometer, interval_ms_);
+    magnetometer_sensor_ = magnetometer_sensor.get();
     magnetometer_channel->SetStreamHandler(
-        std::make_unique<DeviceSensorStreamHandler>(SensorType::kMagnetometer));
+        std::make_unique<DeviceSensorStreamHandler>(
+            std::move(magnetometer_sensor)));
+  }
+
+  void HandleMethodCall(const FlMethodCall &method_call,
+                        std::unique_ptr<FlMethodResult> result) {
+    const std::string &method_name = method_call.method_name();
+
+    if (method_name == "setAccelerationSamplingPeriod") {
+      if (!SetInterval(method_call, result.get())) {
+        return;
+      }
+      if (accelerometer_sensor_) {
+        accelerometer_sensor_->SetInterval(interval_ms_);
+      }
+    } else if (method_name == "setGyroscopeSamplingPeriod") {
+      if (!SetInterval(method_call, result.get())) {
+        return;
+      }
+      if (gyroscope_sensor_) {
+        gyroscope_sensor_->SetInterval(interval_ms_);
+      }
+    } else if (method_name == "setUserAccelerometerSamplingPeriod") {
+      if (!SetInterval(method_call, result.get())) {
+        return;
+      }
+      if (user_accelerometer_sensor_) {
+        user_accelerometer_sensor_->SetInterval(interval_ms_);
+      }
+    } else if (method_name == "setMagnetometerSamplingPeriod") {
+      if (!SetInterval(method_call, result.get())) {
+        return;
+      }
+      if (magnetometer_sensor_) {
+        magnetometer_sensor_->SetInterval(interval_ms_);
+      }
+    } else {
+      result->NotImplemented();
+      return;
+    }
+    result->Success();
+  }
+
+  bool SetInterval(const FlMethodCall &method_call, FlMethodResult *result) {
+    const auto *sampling_period = std::get_if<int32_t>(method_call.arguments());
+    if (!sampling_period) {
+      result->Error("Invalid argument", "No sampling period provided.");
+      return false;
+    }
+
+    if (*sampling_period == 0) {
+      interval_ms_ = 1000;
```

Comment:
What's the reason for this? 1000 is neither the default interval nor the fastest interval.

Reply:
oh It was my mistake.
According to the [documentation](https://developer.tizen.org/dev-guide/tizen-iot-headed/4.0/group__CAPI__SYSTEM__SENSOR__LISTENER__MODULE.html#ga32f88b4d08ae4c373dd83648a05613f6), bound is 10 to 1000, although it may vary for each sensor.
I will fix it to 10.

Reply:
done